server.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const net = require('net');
+const server = net.createServer();
+const parse = require('./lib/parse-message.js');
+
+let sockets = [];
+
+server.on('connection', function(socket) {
+  console.log('someone is here');
+  socket.write('what up, what you got to say');
+  socket.username = `talker_${Math.floor(Math.random() * 100)}`;
+  sockets.push(socket);
+  socket.on('data', function(buffer) {
+    let message = buffer.toString();
+
+    if(message.startsWith('/change username'))
```

Comment:
Having the two words of ``/change username`` messes up some of the functionality. 

I.e., if I say ``change username judy`` my username is now ``username judy`` instead of just ``judy`` as it should be. And this causes your``/dm`` and ``/ban`` commands to not work. 

Back in your ``parse-message.js`` module, we do lots of string manipulations with ``.split, .slice, .join``. 

All of those methods are VERY particular string manipulation methods so that when you type in ``/yourcommand``, the `/` command is not included in the displayed message, and the chatroom only accounts for the messages starting one space after ``/command``. 

I.e., think of a string as an array. 
```
/change username judy
```

The above command would be an array of 3 elements, where ``/change`` is at ``index[0]`` and ``judy`` is at ``index[2]``. 

Line 6 of your ``usernameCommand`` function back in ``parse-message.js``, at the most general explanation, basically copies that string starting at ``index[1]``, which in this case, starts at the word ``username``, therefore, a user has a two-word username, in this case, ```username judy``. 

At the most basic explanation, your ``dmCommand`` and ``banCommand`` functions depend on this username to ***only*** be one word, or else they won't properly detect a username the right way if it's two words. 

This is a long-winded explanation that your ``/change username`` command needs to be one word for your stuff to work. :) But I want you to understand why that is. 

I recommend taking the time to look up all the string manipulation/array methods you see here, e.g. ``split, trim, slice, join``. They'll help in understanding the particulars.